R/pkg/NAMESPACE

```diff
@@ -6,10 +6,15 @@ importFrom(methods, setGeneric, setMethod, setOldClass)
 #useDynLib(SparkR, stringHashCode)
 
 # S3 methods exported
+export("sparkR.session.getOrCreate")
```

Comment:
there is very limited use for SparkContext in R - do you really think we should keep/expose it?
and most of what you are suggesting is already implemented as such :) I think except sparkR.init() is deprecated (but still works for back compatibility)